Putting SDK team knowledge into docs (#8399)
getsentry/sentry-docs: src/platforms/common/configuration/options.mdx

This function is called with an SDK-specific message or error event object, and can return a modified event object, or `null` to skip reporting the event. This can be used, for instance, for manual PII stripping before sending.

> Please, note when <PlatformIdentifier name="before-send" /> is being executed, all scope data has already been applied to the event. Further modification of the scope won't have any effect.

Reviewer: ```suggestion
By the time <PlatformIdentifier name="before-send" /> is executed, all scope data has already been applied to the event. Further modification of the scope won't have any effect.
```
